tests/unit/test_fracture_network.py

```diff
@@ -1,16 +1,115 @@
+        """Check that a type error is raised if the list of fractures contain
+        non-homogeneous types, e.g., it should be all PlaneFractures or
+        LineFractures."""
+        for fractures in fractures_lists:
+            with pytest.raises(TypeError) as excinfo:
+                msg = "All fracture objects must be of the same type."
+                pp.create_fracture_network(fractures, None)
+            assert msg in str(excinfo.value)
+
+    # -----> Test #3
+    @pytest.mark.parametrize("domain", [unit_domain(2), unit_domain(3)])
+    def test_empty_list_and_none_create_the_same_network(self, domain):
+        """Check that is the same giving fractures as an empty list or None."""
+        fn_none = pp.create_fracture_network(None, domain)
+        fn_empty_list = pp.create_fracture_network([], domain)
+        assert fn_none.num_frac() == fn_empty_list.num_frac()
+        assert fn_none.domain == fn_empty_list.domain
+
+    # -----> Test #4
+    def test_warning_raised_if_run_checks_true_for_dim_not_3(self):
+        """Checks if the warning is properly raised when ``run_checks=True`` for a
+        fracture network with a dimensionality different from 3."""
+        warn_msg = "'run_checks=True' has no effect if dimension != 3."
+        with pytest.warns() as record:
+            pp.create_fracture_network(None, domain=unit_domain(2), run_checks=True)
+        assert str(record[0].message) == warn_msg
+
+    # -----> Test #5
+    @pytest.fixture(scope="function")
+    def fractures_list_2d(self, line_fracture):
+        list_1 = []
+        list_2 = [line_fracture]
+        return [list_1, list_2]
+
+    def test_create_fracture_network_2d(self, fractures_list_2d):
+        """Test if 2d fracture networks constructed with FractureNetwork2d and with
+        create_fracture_network() are the same."""
+        for fractures in fractures_list_2d:
+            fn_with_function = pp.create_fracture_network(fractures, unit_domain(2))
+            fn_with_class = FractureNetwork2d(fractures, unit_domain(2))
+            assert fn_with_function.num_frac() == fn_with_class.num_frac()
+            assert fn_with_function.domain == fn_with_class.domain
+
+    # -----> Test #6
+    @pytest.fixture(scope="function")
+    def fractures_list_3d(self, plane_fracture):
+        list_1 = []
+        list_2 = [plane_fracture]
+        return [list_1, list_2]
+
+    def test_create_fracture_network_3d(self, fractures_list_3d):
+        """Test if 3d fracture networks constructed with FractureNetwork3d and with
+        create_fracture_network() are the same."""
+        for fractures in fractures_list_3d:
+            fn_with_function = pp.create_fracture_network(fractures, unit_domain(3))
+            fn_with_class = FractureNetwork3d(fractures, unit_domain(3))
+            assert fn_with_function.num_frac() == fn_with_class.num_frac()
+            assert fn_with_function.domain == fn_with_class.domain
```

Comment:
Don't know if this is needed, but maybe assert that `fn_with_function` is indeed an
instance of `FractureNetwork3d` to check that the domain was correctly inferred.
```suggestion
            assert isinstance(fn_with_function, FractureNetwork3d)
```

Reply:
Incorporated.